modules/search-service-impl/src/main/java/org/opencastproject/search/endpoint/SearchRestService.java

```diff
@@ -465,35 +457,33 @@ public Response getEpisode(
 
     boolean invalidSeries = false;
     if (seriesName != null) {
-      if (seriesService == null) {
-        return Response.status(Response.Status.NOT_IMPLEMENTED).build();
-      }
-      DublinCoreCatalogList result;
+      SearchResult result = new SearchResultImpl();
       try {
-        result = seriesService.getSeries(new SeriesQuery().setSeriesTitle(seriesName));
-      } catch (SeriesException e) {
+        SearchQuery seriesSearch = new SearchQuery();
+        seriesSearch.includeSeries(true)
+                    .includeEpisodes(false)
+                    .withText(seriesName)
+                    .withLimit(1);
+        result = searchService.getByQuery(seriesSearch);
+      } catch (SearchException e) {
         return Response.status(Response.Status.INTERNAL_SERVER_ERROR)
             .entity("Error while searching for series")
             .build();
       }
       // Specifying a nonexistent series ID is not an error, so a nonexistent
       // series name shouldn't be, either.
-      if (result.getTotalCount() == 0) {
+      if (result.getTotalSize() == 0) {
+        logger.debug("Retrieved 0 series results");
         invalidSeries = true;
       } else {
-        if (result.getTotalCount() > 1) {
+        if (result.getTotalSize() > 1) {
           return Response.status(Response.Status.BAD_REQUEST)
               .entity("more than one series matches given series name")
               .build();
         }
```

Comment:
Since limit == 1, will total size ever be bigger than 1?

Reply:
It could, the limit == 1 just limits the number of results returned. The totalSize is the total number of hits on the index, regardless of the number of results returned.